src/lib/motion_planning/PositionSmoothing.cpp

```diff
@@ -106,16 +106,33 @@ float PositionSmoothing::_getMaxXYSpeed(const Vector3f(&waypoints)[3]) const
 
 float PositionSmoothing::_getMaxZSpeed(const Vector3f(&waypoints)[3]) const
 {
+	const Vector3f &start_position = waypoints[0];
+	const Vector3f &target = waypoints[1];
+	const Vector3f &next_target = waypoints[2];
 
-	const auto &target = waypoints[1];
+	const Vector2f start_position_xy_z = {start_position.xy().norm(), start_position(2)};
```

Comment:
good catch, yea youre right. above there is an comment that its done because the position setpoint can been modified by an external source (eg. Obstacle Avoidance)